### ISSUE TITLE
Keyboard shotcut Z/Y swapped on Gemini QWERTZ keyboard

### ISSUE THREAD

**Issue by jirikeller** [thunderbird/thunderbird-android#3529]
In text editing, QWERTZ keyboard works fine, however keyboard shortcuts Y and Z are swaped.

### Expected behavior
If I press "z" button (on QW... row) a read flag should be toggled
If I press "y" button (on YXC.. row) a "copy message" function should be evoked

### Actual behavior
If I press "z" button (on QW... row) a "copy message" function is evoked
If I press "y" button (on YXC.. row) a  read flag is toggled

### Steps to reproduce
1. Switch to QWERTZ keyboard (Czech, German, ...)
2. press the Z and Y keys

### Environment
K-9 Mail version: 5.403 (from F-droid)
Android version: 7.1.1
Account type (IMAP, POP3, WebDAV/Exchange): IMAP, but irrelevant

**Comment by cketti** [comment]
We should probably map the key code to a character before checking which action to trigger. Looks like [`KeyCharacterMap`](https://developer.android.com/reference/android/view/KeyCharacterMap) can be used for this.

The keyboard shortcut handling code lives in `MessageList.onCustomKeyDown()`. https://github.com/k9mail/k-9/blob/378c90b46ee51be85ce11b9f84da497d0be48b96/app/ui/legacy/src/main/java/com/fsck/k9/activity/MessageList.java#L753

**Comment by gbentz** [comment]
I have experimented with a [Unihertz Titan](https://www.unihertz.com/titan.html) running Android 10.

I augmented the `MessageList.onCustomKeyDown()` method as suggested to use the `KeyEvent.getKeyCharacterMap()` and then the `KeyCharacterMap.get()` method to retrieve the unicode character for the `keyCode`.
I find that regardless of which keyboard language is active, the `keyCode` maps to the QWERTY layout unicode character.

It am wondering if the `InputMethodService` performs the correct mapping only when a text input field is active.
The `MessageList` has no active text input field, so perhaps there is no localized handling of `keyCode` values being performed.

This is just a guess, and I would be happy to investigate further if other suggestions are available.
